spec/controllers/sessions_controller_spec.rb

```diff
@@ -29,6 +29,23 @@
         expect(flash[:notice]).not_to be_blank
       end
     end
+
+    context "when authentication passes but account is inactive" do
+      let(:user) { Fabricate(:user, active: false) }
+      before { post :create, email_address: user.email_address, password: user.password }
+
+      it "does not save user id in session" do
+        expect(session[:user_id]).to be_nil
+      end
+
+      it "redirect to login page" do
+        expect(response).to redirect_to login_path
+      end
+
+      it "sets flash error" do
+        expect(flash[:error]).to eq("Your account is currently inactive. An email has been sent to your registered address.")
```

Comment:
Checking exact error message in Controller is OK, but the message can change and its usually better to just check for a presence to have less error prone specs.